modules/social_features/social_group/modules/social_group_request/social_group_request.module

```diff
@@ -299,29 +300,32 @@ function social_group_request_preprocess_activity(array &$variables) {
   $activity = $variables['elements']['#activity'];
 
   if (!$activity->get('field_activity_message')->isEmpty()) {
+    $message_storage = \Drupal::entityTypeManager()->getStorage('message');
     /** @var Drupal\message\Entity\Message $message */
-    $message = $activity->field_activity_message->entity;
+    $message = $message_storage->load($activity->get('field_activity_message')->getValue()[0]['target_id']);
```

Comment:
Please check if `$message_storage->load($activity->get('field_activity_message')->getValue()` returns a value. Then add defensive checks like `isset` for `[0]['target_id'];`

Reply:
@navneet0693 Why we should add `isset` if we have this check above `if (!$activity->get('field_activity_message')->isEmpty()) { .... }`?

Reply:
@Ressinel in case structure changes in future upgrades/updates or for any reason `[0]` is not present, and then this will be a fail-safe.

Reply:
@navneet0693 But if the field is not empty then we will have `[0]` always, no?

Reply:
@navneet0693 @Ressinel I checked and changed check to radio button, also I left $activity->field_activity_message->entity; I think this will be the best option